### PR TITLE
Add linear friction functor material

### LINKED ISSUE #24702: Add a functor material that depends linearly on interstitial speed
## Reason

Currently, it is difficult to represent Forchheimer terms in NSFV: `F * |v_I|^2`. 
`INSFVMomentumFriction` does that, but it doesn't work well with the FVAction
and requires accounting for porosity in the quadratic coefficient.

## Design

Add a `functormaterial`.

## Impact

In the mid-term we want to move to multiply the Forchheimer coefficient with 
`v^2` in `PINSFVMomentumFriction`. We have not moved to this correct definition
because of legacy materials in Pronghorn that all need to be changed (and nobody has
time to do it).

### PR DESCRIPTION
**Pull request by snschune**
closes #24702 

## Reason

It should be easy to define a quadratic loss in PINSFV.

## Design

A functormaterial that does that.

## Impact

see #24702

### CODE REVIEW

File: modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_x, 0, "The x component of the fluid superficial velocity variable.");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_y, 0, "The y component of the fluid superficial velocity variable.");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_z, 0, "The z component of the fluid superficial velocity variable.");
+  params.addParam<RealVectorValue>(
+      "A", RealVectorValue(1, 1, 1), "Coefficient of the A * f(t) term");
+  params.addParam<RealVectorValue>(
+      "B", RealVectorValue(1, 1, 1), "Coefficient of the B * g(t) * |v_I| term");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C:35]: no defaults. or defaults of 0

too risky if a user wants linear and we re adding them a quadratic by default

File: modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", LinearFrictionFactorFunctorMaterial);
+
+InputParameters
+LinearFrictionFactorFunctorMaterial::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params.addClassDescription("Material class used to compute a friction factor of the form A * "
+                             "f(t) + B * g(t) * |v_I| with A, B vector constants, f(t) and g(t) "
```
Comment (GiudGiud) [inline modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C:20]:
```suggestion
                             "f(r, t) + B * g(r, t) * |v_I| with A, B vector constants, f(r, t) and g(r, t) "
```

File: modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                             "functions of time, and |v_I| the interstitial speed");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C:21]:
```suggestion
                             "functors of space and time, and |v_I| the interstitial speed");
```

File: modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/tests [hunk 4]
```diff
@@ -187,4 +187,13 @@
     abs_zero = 1e-9
     issues = '#18478'
   []
+  [linear_friction]
+    type = 'Exodiff'
+    input = linear_friction.i
+    exodiff = linear_friction_out.e
+    issues = '#24703'
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/tests:194]:
```suggestion
    issues = '#24702'
```

File: modules/navier_stokes/doc/content/source/functormaterials/LinearFrictionFactorFunctorMaterial.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+!syntax description /FunctorMaterials/LinearFrictionFactorFunctorMaterial
+
+This class allows to compute the friction coefficient $W$ that is used
+as:
+
+!equation
+-\epsilon \nabla p = W \rho \vec{v}_I
+
+in the porous flow equations where all symbols have their common meaning. The
```
Comment (GiudGiud) [inline modules/navier_stokes/doc/content/source/functormaterials/LinearFrictionFactorFunctorMaterial.md:11]:
```suggestion
in the porous flow equations where all symbols have their common meaning (see [porous media equations](navier_stokes/pinsfv.md)) and we do not write all the other momentum equation terms. The
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]

File: modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        ADReal speed;
+        if ((MooseUtils::absoluteFuzzyEqual(_superficial_vel_x(r, t), 0)) &&
+            (MooseUtils::absoluteFuzzyEqual(_superficial_vel_y(r, t), 0)) &&
+            (MooseUtils::absoluteFuzzyEqual(_superficial_vel_z(r, t), 0)))
+          speed = 1e-42;
+
+        speed = ADRealVectorValue(
+                    _superficial_vel_x(r, t), _superficial_vel_y(r, t), _superficial_vel_z(r, t))
+                    .norm() /
+                _eps(r, t);
```
Comment (lindsayad) [inline modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C:80]: I would save off a local velocity variable and then call `NS::computeSpeed`
```suggestion
        const ADRealVectorValue superficial_vel(_superficial_vel_x(r, t),
                                                _superficial_vel_y(r, t),
                                                _superficial_vel_z(r, t));
        const auto speed = NS::computeSpeed(superficial_vel) / _eps(r, t);
```

**Review by lindsayad — DISMISSED** [review]

File: modules/navier_stokes/doc/content/source/functormaterials/LinearFrictionFactorFunctorMaterial.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# LinearFrictionFactorFunctorMaterial
+
+!syntax description /FunctorMaterials/LinearFrictionFactorFunctorMaterial
+
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/functormaterials/LinearFrictionFactorFunctorMaterial.md:4]: I would probably just remove these lines as it introduces the equation without formatting whereas you have it with formatting later in this file

File: modules/navier_stokes/include/functormaterials/LinearFrictionFactorFunctorMaterial.h [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+  LinearFrictionFactorFunctorMaterial(const InputParameters & parameters);
+
+protected:
+  /// name of the functor computed by this material
+  MooseFunctorName _functor_name;
+
+  ///@{ A, B, f(t), g(t)
+  RealVectorValue _A;
+  RealVectorValue _B;
```
Comment (lindsayad) [inline modules/navier_stokes/include/functormaterials/LinearFrictionFactorFunctorMaterial.h:32]:
```suggestion
  const RealVectorValue _A;
  const RealVectorValue _B;
```

File: modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/linear_friction.i [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+[Modules]
+  [NavierStokesFV]
+    compressibility = 'incompressible'
+    porous_medium_treatment = true
+
+    density = 'rho'
+    dynamic_viscosity = 'mu'
+    porosity = 'porosity'
+
+    initial_velocity = '1 1e-6 0'
```
Comment (lindsayad) [inline modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/linear_friction.i:24]: We probably don't need these small perturbations from a 0 initial condition. We may have needed them at some point, but I hope we don't still

**Review by lindsayad — COMMENTED** [review]
Just small comments

**Comment by lindsayad** [comment]
@snschune I will fix the exodiff and squash some of the commit history

**Review by lindsayad — APPROVED** [review]

**Review by GiudGiud — APPROVED** [review]
looks good!

**Pull request merged**